vit_pytorch/vit_pytorch.py

```diff
@@ -109,10 +109,13 @@ def __init__(self, *, image_size, patch_size, num_classes, dim, depth, heads, ml
             nn.Linear(dim, num_classes)
         )
 
-    def forward(self, img, mask = None):
-        p = self.patch_size
+    def forward(self, img, mask = None, mpp = False):
```

Comment:
i think you can actually avoid modifying ViT altogether if you just reach into the ViT class and invoke the `.transformer`

Reply:
I'm not sure if this is right, but doesn't the transformer take the embeddings of inputs after they have gone through a linear transformation and with the addition of the cls token. If I were to just invoke `.transformer` from the wrapper, I think the linear transformation and cls token would have to be parameters of the mpp wrapper module.

Reply:
ohh yea, you are right... hmm I think you should just reach in and grab the cls token and linear transform as well. the benefits of leaving the base class unchanged outweighs the extra complexity in the wrapper, imo